### ISSUE TITLE
Towards making SAP our default discrete solver

### ISSUE THREAD

**Issue by amcastro-tri** [RobotLocomotion/drake#19322]
This Epic tracks the development of code that, after our recent research, we know how to write. We believe these are the last missing bits towards making SAP the default solver in Drake.

- [x]  #19434.
- [x]  #19320.
- [x]  #19435.
- [x]  Implicit PD controllers, #19067.
- [ ]  Joint stiction #18932 
- [ ]  Document solver failures modes (whether internal or as in most cases, external). Consider an entry in the [Troubleshooting](https://drake.mit.edu/troubleshooting.html) page in Drake.

**Comment by aminya** [comment]
One issue with the SapSolver is the `d2ellR_dalpha2` range failure. This line sometimes fails for me, but I cannot gain insight into the reasons.

https://github.com/RobotLocomotion/drake/blob/c482ed7f6222958eb1f2904db9bbbd22e6eedf9f/multibody/contact_solvers/sap/sap_solver.cc#L341

**Comment by amcastro-tri** [comment]
Hi @aminya, we actually run into this issue ourselves last week.
The underlying reason was not the solver really but there was garbage (large values, order of 1e300) in the inputs. Since SAP is the only place where numerics is checked, that's were the code first barks. 
I wonder if you had a similar experience. In my dev branch I put a check for those large values, probably worth pushing into master if a common problem. WDYT?

**Comment by aminya** [comment]
@amcastro-tri I verified that there are no garbage or large values in the inputs. I disabled the three checks to see what happens. But it fails at this line randomly:

```cpp
newton_with_bisection/drake/multibody/contact_solvers/newton_with_bisection.h:31 in Bracket(): condition 'has_different_sign(f_lower, f_upper)' failed.
```

This is random and it doesn't happen all the time. It's very hard to have a reproducible setup.  

The only reason I am using the SAP solver now is the mimic joint support. Is there an alternative solution for me?
